src/modm/platform/gpio/rp/base.hpp.in

```diff
@@ -133,8 +148,52 @@ struct Gpio::PortRegs<Gpio::Port::{{ port | capitalize }}>
 			PADS_BANK0_GPIO0_SLEWFAST_BITS
 		);
 	}
+	static void enable_irq(uint8_t pin, InputTrigger triggers)
+	{
+		uint32_t value = static_cast<uint32_t>(triggers) << 4 * {{intreg_pin_mask_shift[port]}};
+%% if multicore_enabled
+		auto& proc_irq_ctrl = sio_hw->cpuid ? io{{port | lower}}_hw->proc1_{{intreg_ctrl_names[port]}} : io{{port | lower}}_hw->proc0_{{intreg_ctrl_names[port]}};
+%% else
+		auto& proc_irq_ctrl = io{{port | lower}}_hw->proc0_{{intreg_ctrl_names[port]}};
+%% endif
+		hw_set_bits(&proc_irq_ctrl.inte{{intreg_pin_access[port]}}, value);
+	}
+	static void disable_irq(uint8_t pin, InputTrigger triggers)
+	{
+		uint32_t value = static_cast<uint32_t>(triggers) << 4 * {{intreg_pin_mask_shift[port]}};
+%% if multicore_enabled
+		auto& proc_irq_ctrl = sio_hw->cpuid ? io{{port | lower}}_hw->proc1_{{intreg_ctrl_names[port]}} : io{{port | lower}}_hw->proc0_{{intreg_ctrl_names[port]}};
+%% else
+		auto& proc_irq_ctrl = io{{port | lower}}_hw->proc0_{{intreg_ctrl_names[port]}};
+%% endif
+		hw_clear_bits(&proc_irq_ctrl.inte{{intreg_pin_access[port]}}, value);
+	}
+	static void acknowledge_irq(uint8_t pin, InputTrigger triggers)
+	{
+		uint32_t value = static_cast<uint32_t>(triggers) << 4 * {{intreg_pin_mask_shift[port]}};
+		io{{port | lower}}_hw->intr{{intreg_pin_access[port]}} = value;
+	}
 };
+
 %% endfor
 /// @endcond
 
+inline constexpr Gpio::InputTrigger operator|(Gpio::InputTrigger a, Gpio::InputTrigger b)
+{
+	using enum_ut = std::underlying_type_t<Gpio::InputTrigger>;
+	return static_cast<Gpio::InputTrigger>(static_cast<enum_ut>(a) | static_cast<enum_ut>(b));
+}
+
+inline constexpr Gpio::InputTrigger operator&(Gpio::InputTrigger a, Gpio::InputTrigger b)
+{
+	using enum_ut = std::underlying_type_t<Gpio::InputTrigger>;
+	return static_cast<Gpio::InputTrigger>(static_cast<enum_ut>(a) & static_cast<enum_ut>(b));
+}
```

Comment:
The [register module](https://modm.io/reference/module/modm-architecture-register/) takes care of this via `MODM_FLAGS` macro.